### PR TITLE
Apply and reset "more" filters

### PR DESCRIPTION
**Pull request by monkemedia**

### CODE REVIEW

File: components/search/SearchInterface.vue [hunk 1]
```diff
@@ -238,6 +240,27 @@
...
         return ordered.concat(unordered);
       },
+      coreFacets() {
+        return this.orderedFacets.filter(facet => ['TYPE', 'COUNTRY', 'REUSABILITY'].includes(facet.name));
+      },
+      moreFacets() {
+        const coreFacetNames = this.coreFacets.map(facet => facet.name);
+        return this.orderedFacets.filter(facet => !coreFacetNames.includes(facet.name));
+      },
+      moreSelectedFacets() {
```
Comment (rwd) [inline components/search/SearchInterface.vue]: Let's use a lodash function for this:
```diff
diff --git a/components/search/SearchInterface.vue b/components/search/SearchInterface.vue
index 57096ed..9f27cf7 100644
--- a/components/search/SearchInterface.vue
+++ b/components/search/SearchInterface.vue
@@ -139,6 +139,7 @@
   import { defaultFacets } from '../../plugins/europeana/search';
 
   import isEqual from 'lodash/isEqual';
+  import omitBy from 'lodash/omitBy';
   import { mapState } from 'vuex';
 
   export default {
@@ -177,6 +178,11 @@
         default: Boolean(Number(process.env['ENABLE_MORE_FACETS']))
       }
     },
+    data() {
+      return {
+        coreFacetNames: ['TYPE', 'COUNTRY', 'REUSABILITY']
+      };
+    },
     computed: {
       ...mapState({
         error: state => state.search.error,
@@ -241,25 +247,13 @@
         return ordered.concat(unordered);
       },
       coreFacets() {
-        return this.orderedFacets.filter(facet => ['TYPE', 'COUNTRY', 'REUSABILITY'].includes(facet.name));
+        return this.orderedFacets.filter(facet => this.coreFacetNames.includes(facet.name));
       },
       moreFacets() {
-        const coreFacetNames = this.coreFacets.map(facet => facet.name);
-        return this.orderedFacets.filter(facet => !coreFacetNames.includes(facet.name));
+        return this.orderedFacets.filter(facet => !this.coreFacetNames.includes(facet.name));
       },
       moreSelectedFacets() {
-        // Strips out facets which don't belong to the More Facets dropdown
-        const obj = {};
-        this.moreFacets.map(facet => {
-          console.log(facet);
-          for (let key in this.selectedFacets) {
-            if (facet.name === key) {
-              obj[key] = this.selectedFacets[key];
-            }
-          }
-        });
-
-        return obj;
+        return omitBy(this.selectedFacets, (selected, name) => this.coreFacetNames.includes(name));
       },
       showPagination() {
         return this.totalResults > this.perPage;
```

File: components/search/MoreFacetsDropdown.vue [hunk 2]
```diff
@@ -55,24 +73,89 @@
...
       return {
-        selected: {}
+        preSelected: {}
       };
     },
 
     computed: {
-      selectedAmount() {
-        return [].concat(...Object.values(this.selected)).length;
+      preSelectedOptions() {
```
Comment (rwd) [inline components/search/MoreFacetsDropdown.vue]: This does not appear to be used.

File: components/search/MoreFacetsDropdown.vue [hunk 3]
```diff
@@ -55,24 +73,89 @@
...
+        return [].concat(...Object.values(this.preSelected));
+      },
+
+      selectedOptions() {
```
Comment (rwd) [inline components/search/MoreFacetsDropdown.vue]: This is only ever used to count the options, so let's:
```diff
diff --git a/components/search/MoreFacetsDropdown.vue b/components/search/MoreFacetsDropdown.vue
index 41f5b9f..984d159 100644
--- a/components/search/MoreFacetsDropdown.vue
+++ b/components/search/MoreFacetsDropdown.vue
@@ -4,17 +4,17 @@
     variant="light"
     no-caret
     class="more-facets position-static"
-    :class="{ 'is-active' : selectedOptions.length > 0 }"
+    :class="{ 'is-active' : selectedOptionsCount > 0 }"
     data-qa="more filters dropdown button"
     @hidden="cancelHandler"
   >
     <template v-slot:button-content>
       {{ $t('facets.button.morefilters') }}
       <span
-        v-if="selectedOptions.length > 0"
+        v-if="selectedOptionsCount > 0"
         class="selected-amount"
       >
-        {{ selectedOptions.length }}
+        {{ selectedOptionsCount }}
       </span>
     </template>
     <b-dropdown-group class="more-facets-wrapper">
@@ -33,7 +33,7 @@
     >
       <b-button
         variant="link"
-        :disabled="selectedOptions.length < 1"
+        :disabled="selectedOptionsCount < 1"
         data-qa="reset filter button"
         @click="resetFilters"
       >
@@ -91,8 +91,8 @@
         return [].concat(...Object.values(this.preSelected));
       },
 
-      selectedOptions() {
-        return [].concat(...Object.values(this.selected));
+      selectedOptionsCount() {
+        return Object.values(this.selected).length;
       },
 
       disableButton() {
@@ -114,7 +114,7 @@
       updateSelected(facetName, selectedFields) {
         Vue.set(this.preSelected, facetName, selectedFields);
 
-        if (this.selectedOptions.length === 0) {
+        if (this.selectedOptionsCount === 0) {
           this.clearEmpty();
         }
       },
```

File: components/search/MoreFacetsDropdown.vue [hunk 4]
```diff
@@ -55,24 +73,89 @@
...
+        this.$refs.dropdown.hide(true);
+      },
+
+      resetFilters() {
+        this.clearPreSelected();
+        this.$emit('changed', this.preSelected);
+        this.clearEmpty();
+      },
+
+      clearSaved() {
```
Comment (rwd) [inline components/search/MoreFacetsDropdown.vue]: This is never called.

File: components/search/MoreFacetsDropdownOptions.vue [hunk 5]
```diff
@@ -67,19 +68,41 @@
       name: {
         type: String,
         required: true
+      },
+      preSelected: {
```
Comment (rwd) [inline components/search/MoreFacetsDropdownOptions.vue]: This seems to follow the opposite naming convention as the prop/data in MoreFacetsDropdown.vue. Is that accurate? If so, please align them.

File: components/search/MoreFacetsDropdownOptions.vue [hunk 6]
```diff
@@ -67,19 +68,41 @@
...
     },
 
+    watch: {
+      preSelected(value) {
+        this.selected = value;
+      }
+    },
+
+    mounted() {
+      this.$root.$on('updateSelectedOptions', this.updateOptions);
```
Comment (rwd) [inline components/search/MoreFacetsDropdownOptions.vue]: This is not the way to have parent components communicate data changes to children. Refactor to rely on the props changing.

File: components/search/MoreFacetsDropdown.vue [hunk 7]
```diff
@@ -54,24 +76,71 @@
...
+    },
+
+    watch: {
+      selected() {
+        this.preSelected = this.cloneSelected();
       }
     },
 
     methods: {
+      cloneSelected() {
```
[the remark this answers is not in the document]
Comment (rwd) [reply components/search/MoreFacetsDropdown.vue]: This is making a copy of the original selections, so this name is fine.

[the remark this answers is not in the document]
Comment (lbiedinger) [reply components/search/MoreFacetsDropdown.vue]: ignore this

**Pull request merged**